src/tablet/tablet_impl.cc

```diff
@@ -1036,7 +1037,8 @@ int32_t TabletImpl::ScanIndex(const ::openmldb::api::ScanRequest* request, const
             continue;
         }
         uint64_t ts = combine_it->GetTs();
-        if (ts <= et) {
+        // et is inclusive
+        if (ts < et) {
```

Comment:
special handling for `et == 0` only to make it compatible.

Reply:
should be revert in #2669